### ISSUE TITLE
HCP Meeting Notes

### ISSUE THREAD

**Issue by saltzmanj** [saltzmanj/keyboardguys#8]
## Monday, September 12th
- LabVIEW has camera interfaces: image datatype
  - include large palette of tools, but not all can go on myRIO
  - can be emulated on PC, if the elements are available also on myRIO
- Machine Learning Toolkit for LabVIEW available on NI Website
- Python window also available 
- **Image Recognition**
  - Print out music at high contrast
  - Avoid glare: backlighting
    - usually works well for binary pixel classification
  - Tools available to map shapes to each other
  - Vision library is pretty huge
  - "Blob analysis": very potent tool
    - can define custom threshold
  - Need to think out fiduciary marks
  - Investigate lensing of cameras
  - Find cameras that work well with myRIO
    - Telecentric lens: probably too expensive
  - First functions for processing: find a region of interest
    - Next: resizing
- **Gantt Chart**
  - Need to allow time for synthesis and testing: make that a new task(s)
  - Keep in mind this is a living document
  - Add sub-tasks, such as for PCBs
    - Include testing times
- **PCB/Hardware**
  - Come up with a testing plan, order a few before all 44+
    - Include this in the Gantt
  - Think about power and signal connections, feedback
  - Needs a test sequence
- After lab reveal: test plans

**Comment by saltzmanj** [comment]
## Monday, September 19th

- share Gantt file on Collab
- machine learning library in LabVIEW which may be helpful
     - Python or Mathscript also, but will be slower than native LabVIEW
- Overall, proposal looks great

**Comment by saltzmanj** [comment]
# Monday, September 26
- Optimal minimization strategy
  - Look carefully at myRIO vision libraries, SVMs
  - binary images
- Image processing work
   1. Find staff locations
   2. Look for vertical lines representing the boundary of each measure
   3. Check to ensure sum of notes is consistent for time signature
- Starting out with music notes
  1. Get images of music with cell phone, in-focus, well lit
  2. Simulate camera acquisition
  3. Explore high-level functions in image processing library
  4. Experiment: what image processing functions work 
- May need to set limit for ledger lines 
- Can use transceiver for interfacing differential signaling with LED driver
  - RS/EIA-422, RS/EIA-485: differential signaling standards
- Driver board
   - Include regulator: 5V/3.3V
   - Use a driver board for each octave
   - On last board: add shunt jumper so data can shift out to myRIO
      - System can detect number of octaves
   - Use through-hole LEDs: flexible cable to jumper headers to connect LEDs to driver boards
- Check out laser harp
- Maybe: use 24-pin ribbon cable to connect octave boards
- Think about note feedback
- RGB LEDs probably have 3 wires
- High efficiency LEDs: probably only one color, but only uses ~2 mA 
- Don't connect power supplies in parallel!

**Comment by MrBitSage** [comment]
# Monday, October 10
- Powell has emailed Kira about NI Licenses
 - M80X85770 Not valid
- Consider 20 pin ribbon cable and make half of them ground
- Investigate adequate current supply from 9V to mitigate wasted wattage / heat
- Look into switching regulators instead of linear regulators (3 terminal drop in replacements for linear)
 - Also, use regulators on each board instead of just the first one
- Get + and - on opposite pins so that they're adjacent and have 2 grounds between each set of +-s
- PCB Fabrication turnaround is about a week
- No glare problem so don't need backlights
- Use IMAQ instead of Vision Assistant because Vision Assistant may not work on myRIO
-Deliverables for next week
 - Have fiducial recognition and rotational alignment done using IMAQ
 - Gerber files for PCB
 - Implemented OPTICS Clustering
